aten/src/ATen/native/mps/operations/Sorting.mm

```diff
@@ -0,0 +1,80 @@
+#include <ATen/ATen.h>
+#include <ATen/Tensor.h>
+#include <ATen/Utils.h>
+#include <ATen/mps/MPSStream.h>
+#include <ATen/native/mps/OperationUtils.h>
+#include <torch/library.h>
+#include <c10/util/Optional.h>
+#include <ATen/native/mps/MPSGraphVenturaOps.h>
+
+namespace at {
+namespace native {
+
+Tensor argsort_stable_mps(const Tensor & self, bool stable, int64_t dim, bool descending) {
+
+  if(!is_macos_13_or_newer()){
+        TORCH_WARN_ONCE("MPS: median op is supported natively starting from macOS 13.0. ",
+                    "Falling back on CPU. This may have performace implications.");
+        return std::get<1>(at::sort(self, stable, dim, descending));
+  }
+
+  using namespace mps;
+  MPSStream* stream = getCurrentMPSStream();
+  Tensor result = std::get<1>(at::sort(self, stable, dim, descending));
```

Comment:
Why are you adding sort here ? if you want to get the shape of output, you just allocate empty tensor as [shown](https://github.com/pytorch/pytorch/pull/94479/files#diff-902691b53a62fd31d6618a155f5760ae9adea34d16435792b71d608a26c9f2edR16)

Reply:
Thanks for your comment! I removed sort and added empty_mps.

https://github.com/pytorch/pytorch/pull/87572/commits/c07c717c0ed6d92e897ecee4e689d9262dcccf25